clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -285,24 +310,24 @@ private EhcacheEntityResponse invokeLifeCycleOperation(ClientDescriptor clientDe
       case DESTROY_SERVER_STORE:
         destroyServerStore(clientDescriptor, (DestroyServerStore) message);
         break;
+      case CREATE_CLUSTERED_MAP:
+        createConcurrentMap(clientDescriptor, (CreateClusteredMap) message);
```

Comment:
Yeah it could be done. Since every clustered map operation message contains the cache name and the map name, we can delay the creation of server side state repository and the map itself till the first operation on the map is performed.

So just to reiterate the approach, when a client performs a map operation and that message reaches the server side, the cache name is extracted and a state repository with that name will be created if one is not found, and then within the retrieved/created state repository the map is created using the extracted map name if one is not found and then the actual operation is performed on that created/retrieved map.